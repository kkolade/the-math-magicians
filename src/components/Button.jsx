import PropTypes from 'prop-types';
import React from 'react';

const Button = ({ className, onClick, children }) => (
  <>
    <button type="submit" className={className} onClick={onClick}>
      {children}
    </button>
  </>
);

Button.propTypes = {
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};

export default Button;
