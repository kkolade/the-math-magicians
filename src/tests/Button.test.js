import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../components/Button';

describe('testing Button', () => {
  test('Button snapshot renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Button component renders correctly', () => {
    render(<Button />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
