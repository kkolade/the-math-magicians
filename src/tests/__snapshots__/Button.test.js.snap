// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button correctly 1`] = `
<button
  class="test-class"
  type="submit"
>
  Test Button
</button>
`;
